src/node/rpc/frontend.h

```diff
@@ -625,20 +629,23 @@ namespace ccf
         catch (RpcException& e)
         {
           ctx->set_error(std::move(e.error));
+          ctx->clear_response_headers();
```

Comment:
```suggestion
          ctx->clear_response_headers();
          ctx->set_error(std::move(e.error));
```

I think that all of these should clear response headers _before_ setting an error, in case `set_error` includes some headers.

Reply:
I'd checked we don't, but you're right that it's safer.